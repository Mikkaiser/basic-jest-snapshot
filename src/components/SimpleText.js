import React, { useState } from 'react';
import { Text, View, Button } from 'react-native';

export default ({ initialName }) => {
    const [name, setName] = useState('Random Name');

    const modifyName = () => {
        setName('Mikkaiser Modified');
    }

    return (
        <View>
            <Text>{initialName} - This is the name</Text>
            <Text>Mikkaiser est√° aqui</Text>
            <Button
                title={name}
                onPress={modifyName}
            />
        </View>
    );
}