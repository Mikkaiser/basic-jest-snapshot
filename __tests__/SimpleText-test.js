import React from 'react';
import renderer from 'react-test-renderer';
import SimpleText from '../src/components/SimpleText';

test('SimpleText renders correctly', () => {
    const tree = renderer.create(<SimpleText initialName={'Mikkaiser'} />).toJSON();
    expect(tree).toMatchSnapshot();
})