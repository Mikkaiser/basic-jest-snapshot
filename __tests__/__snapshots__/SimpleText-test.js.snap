// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleText renders correctly 1`] = `
<View>
  <Text>
    Mikkaiser
     - This is the name
  </Text>
  <Text>
    Mikkaiser est√° aqui
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Random Name
      </Text>
    </View>
  </View>
</View>
`;
